src/Strategy/Go/Gomod.hs

```diff
@@ -218,11 +219,23 @@ parsePackageVersion lexify = parseSemOrPseudo <|> parseNonCanonical
 gomodParser :: Parser Gomod
 gomodParser = do
   _ <- scn
-  _ <- lexeme (chunk "module")
-  name <- modulePath
-  _ <- scn
-  statements <- many (statement <* scn)
-  eof
+  (name, statements) <-
+    choice
+      [ -- Non-empty go.mod file
```

Comment:
[nit] Binding these `do` blocks to descriptive names would accomplish the same thing as this comment and be easier to read IMO. Something like `choice [emptyGoMod, nonEmptyGoMod]`.